level1/recommendNewId.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class getNewId {
+    public static void main(String[] args){
+
+        Scanner scan = new Scanner(System.in);
+
+        String newId = scan.next();
+        String recommendId  = solution(newId);
+
+        System.out.println(recommendId);
+    }
+    
+    
+    public static String solution(String  new_id){
+
+        //1단계
+        String[] new_id_arr = new_id.toLowerCase().split("");
+        String CAPITAL = "[a-z\\d\\_\\-\\.]";
+        //2단계
+        for (int i = 0; i< new_id_arr.length;i++){
+            if (!Pattern.matches(CAPITAL,new_id_arr[i])){
+                new_id_arr[i] = "";
+            }
+        }
+        new_id = Arrays.stream(new_id_arr).collect(Collectors.joining());
+
+        //3단계
+        boolean flag = true;
+
+        while(flag){
+
+            if(new_id.indexOf("..")>-1) {
+                new_id = new_id.replace("..",".");
+                flag = true;
+            }else{
+                flag = false;
+            }
+
+        }
```

Comment:
이부분도 얼핏 보기에는 한번의 반복문으로 처리한걸로 보이지만...
.indexOf 메서드가 자체적으로 loop문을 갖고있어서 이중루프로 처리됐어요

일단 저라면 이런식으로 했을것 같아요!
더 좋은방법 있을테니 다른사람들 푼거 보고 공부해보는것도 좋아요!

```suggestion
int startIdx = -1;
for ( new_id길이만큼 반복 ) {
  if ( startIdx < 0 & 현재문자가 . 인지? ) {
    startIdx = idx;
  } else if ( startIdx > 0 & startIdx+1 != idx & 현재문자가 . 아닌지? ) {
    new_id 잘라주기 startIdx+1부터 idx-1 까지 지우기?
    startIdx = -1; // 초기화
  }
}
// 마지막 idx 한번더 확인필요...
```